Tests/make verify fail

aaperis opened the pull request:
## Related issue(s) and PR(s)
This PR closes #1505 .


## Description


## How to test

jbygdell commented on sda/cmd/verify/verify.go:
```diff
@@ -345,60 +345,56 @@ func main() {
-							continue
-						}
+				switch {
+				case fileInfo.DecryptedChecksum != "" && fileInfo.DecryptedChecksum == fmt.Sprintf("%x", sha256hash.Sum(nil)):
```
```suggestion
				case fileInfo.DecryptedChecksum == fmt.Sprintf("%x", sha256hash.Sum(nil)):
```

The second part can handle all three possible combinations alone.

jbygdell commented on sda/cmd/verify/verify.go:
```diff
@@ -345,60 +345,56 @@ func main() {
 
-							continue
-						}
+				switch {
```
Technically this switch is not needed. If the `db.UpdateFileEventLog` is moved outside it can be simplified to:
```
if  fmt.Sprintf("%x", sha256hash.Sum(nil)) != fileInfo.DecryptedChecksum {
	if err := db.SetVerified(file, message.FileID); err != nil {
		log.Errorf("SetVerified failed, reason: (%s)", err.Error())
		if err := delivered.Nack(false, true); err != nil {
			log.Errorf("failed to Nack message, reason: (%s)", err.Error())
		}

		continue
	}
}			
```

jbygdell reviewed (changes requested):
The changes to the api is not related to the scope of the issue

MalinAhlberg commented on sda/cmd/verify/verify.go:
```diff
@@ -345,60 +345,56 @@ func main() {
 
-						if err := delivered.Ack(false); err != nil {
-							log.Errorf("failed to Ack message, reason: (%s)", err.Error())
+					if err := db.UpdateFileEventLog(message.FileID, "verified", delivered.CorrelationId, "ingest", "{}", string(verifiedMessage)); err != nil {
```
What is the reasoning for updating the event log here..?

aaperis replied:
This came about because of [this addition](https://github.com/neicnordic/sensitive-data-archive/pull/1526/files#r2031455478) which changed the behavior of verify (and also basically the current issue).

MalinAhlberg reviewed (approved):
Good work!
If one wants to keep the extra test I added (2e48725e157340d21010c1ce2e557977e4faf07e) one could update `5` to `6` [here](https://github.com/neicnordic/sensitive-data-archive/blob/2684564e4ad914a218f2792a405c7b8d4aa6ad0b/.github/integration/tests/sda/21_cancel_test.sh#L89). Then the tests pass (on my machine at least :stuck_out_tongue_winking_eye: ).
Anyway, I approve.

aaperis commented:
> Good work! If one wants to keep the extra test I added ([2e48725](https://github.com/neicnordic/sensitive-data-archive/commit/2e48725e157340d21010c1ce2e557977e4faf07e)) one could update `5` to `6` [here](https://github.com/neicnordic/sensitive-data-archive/blob/2684564e4ad914a218f2792a405c7b8d4aa6ad0b/.github/integration/tests/sda/21_cancel_test.sh#L89). Then the tests pass (on my machine at least 😜 ). Anyway, I approve.

Since rebasing to the merged code of #1526, the error that the code you had added changed. Hence it is only meaningful to check if the file is already verified, which is implicitly done in subsequent tests of the workflow. That is why I removed this part :sunglasses: .

aaperis commented on sda/cmd/verify/verify.go:
```diff
@@ -344,51 +344,37 @@ func main() {
-
-					continue
+				} else {
+					log.Warnln("file already verified")
```
There needs to be a way to somehow follow what happens during ingestion. I think this is not superfluous as it is.

jbygdell commented on sda/cmd/verify/verify.go:
```diff
@@ -344,51 +344,37 @@ func main() {
-
-					continue
+				} else {
+					log.Warnln("file already verified")
```
```suggestion
					log.Infof("file with correlation id: %s is already verified", delivered.CorrelationId)
```

nanjiangshu reviewed (approved):
Looks great!
